api/pkg/apis/v1alpha1/providers/stage/patch/patch.go

```diff
@@ -163,6 +163,10 @@ func (i *PatchStageProvider) Process(ctx context.Context, mgrContext contexts.Ma
 
 	switch patchSource {
 	case "", "catalog":
+		if componentName == "" {
```

Comment:
I don't think this is an appropriate place. Like I've mentioned in the comment, componentName could be empty because we allow patches to add components. See line 239.

Reply:
I see. Catalog patch source can also carry ComponentSpec.